### ISSUE TITLE
v69 "result":"undefined", "rid":14

### ISSUE THREAD

**Issue by tjg2256** [r2-studio/robotmon-scripts#538]
![image](https://github.com/user-attachments/assets/0ad19796-b878-4e06-8e05-56b409b59455)

I've been using RobotMon / Tsum script for a while now. I use MuMu12 as my emulator, running Android 12. I consistently get this error when runnin the latest Tsum script. 

If I use older scripts (v52) it works as intended, but the newest versions encounter this error. Is this an error within the script, or a setting in my emulator that I need to adjust? 

Thank you!

**Comment by mcs** [comment]
Hi, can you provide the file `robotmon.log` frome somewhere like `/sdcard/Download/Robotmon` and give some additional information like if there is a recognizable pattern when the error occurs (Receive hearts? Play game? Buy stuff?)

Maybe it is the script which needs to be fixed, maybe the emulator behaves strange. Hopefully we will find out with the log file 😃

**Comment by tjg2256** [comment]
Wow thank you for your swift reply! I've cleared the log, attempted to run again, and reproduced the situation. 

`bolt.Close(): funlock error: function not implemented
function not implemented
Listen :8081 for gRPC
Use :8082 for broadcast
Listen :8080 for http2
New Logger 1
[Result] {"result":"undefined","rid":1}
[Error] Run script error: ReferenceError: 'stop' is not defined
[Result] {"result":"Run script error: ReferenceError: 'stop' is not defined","rid":2}
[Settings] Save settings 
[Result] {"result":"undefined","rid":3}
[Settings] Save settings 
[Result] {"result":"undefined","rid":4}
[Settings] Save settings 
[Result] {"result":"undefined","rid":5}
[Settings] Save settings 
[Result] {"result":"undefined","rid":6}
[Settings] Save settings 
[Result] {"result":"undefined","rid":7}
[Settings] Save settings 
[Result] {"result":"undefined","rid":8}
[Settings] Save settings 
[Result] {"result":"undefined","rid":9}
[Settings] Save settings 
[Result] {"result":"undefined","rid":10}
[Settings] Save settings 
[Result] {"result":"undefined","rid":11}
[Settings] Save settings 
[Result] {"result":"undefined","rid":12}
[Error] Run script error: ReferenceError: 'stop' is not defined
[Result] {"result":"Run script error: ReferenceError: 'stop' is not defined","rid":13}
`

Hope this is insightful.

**Comment by mcs** [comment]
Well, I just installed MuMu 12 myself and had a somewhat hard time starting the script. Then I tried v52 which worked fine so far. After that, I switched back to v69 - and this time it also worked fine. Received hearts and auto played without problems.

What I observe is that the script itself does not have any write permissions to the path where it wants to save settings or create temporary files. This is indeed a problem, and I am not sure how to make it work for MuMu 12. So I would say that this emulator *can* work with Robotmon and the Tsum and TsumBeta script, but just in a limited way. Recording received hearts absolutely needs write access to the Android file system, so that won't work at all.
